src/manifolds/Unitary.jl

```diff
@@ -111,6 +112,45 @@ project(::UnitaryMatrices{TypeParameter{Tuple{1}},ℍ}, p) = sign(p)
 
 project(::UnitaryMatrices{TypeParameter{Tuple{1}},ℍ}, p, X) = (X - conj(X)) / 2
 
+@doc raw"""
+    project(M::UnitaryMatrices,p)
+
+Projects `p` from the embedding onto the [`UnitaryMatrices`](@ref) `M`, i.e. compute `q`
+as the polar decomposition of ``p`` such that ``q^{\mathrm{H}}q`` is the identity,
+where ``⋅^{\mathrm{H}}`` denotes the hermitian, i.e. complex conjugate transposed.
+"""
+project(::UnitaryMatrices, ::Any, ::Any)
+
+function project!(::UnitaryMatrices, q, p)
```

Comment:
The same `project!` is at Manifolds/src/manifolds/GeneralUnitaryMatrices.jl:968

Reply:
Interesting, it was not triggered in the rand so I added it, but maybe I was also tired. we could remove it if the general one works fine.

Reply:
Probably the tangent vector projection was necessary but not the point projection.

Reply:
Ok, I am still busy today, we could try the tests without the point proj then, or for now leave it a is (and at some point spent a bit of time to reduce actual code duplication throughout the package).